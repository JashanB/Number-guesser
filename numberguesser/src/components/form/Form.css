.check-button {
  background-color: aqua;
}

.check-button:hover {
  background-color: wheat;
}