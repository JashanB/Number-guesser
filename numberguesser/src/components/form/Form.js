import React, { useState, useEffect } from "react";

const Button = (props) => {
  return (
    <button onClick={function () {
      if (props.checkAnswer === false) {
        props.setCheckAnswer(state => (true));
        console.log('actual', props.actualAnswer)
        console.log('input', props.inputNumber)

        if (props.actualAnswer == props.inputNumber) {
          console.log('equal')
          props.setCompareAnswer(state => (true));
        } else {
          console.log('not equal')
          props.setCompareAnswer(state => (false));
        }
      }
    }
    }
    >Check!</button>
  )
}

export default function Form(props) {
  return (
    <div className="input-your-guess">
      <form autoComplete="off" onSubmit={event => event.preventDefault()}>
        <input
          className="guess-number"
          type="number"
          placeholder="Guess a number!"
          onChange={(event) => {
            props.setInputNumber(event.target.value)
          }
          }
          value={props.inputNumber}
          data-testid="number-input"
        />
      </form>
      <Button setCheckAnswer={props.setCheckAnswer} checkAnswer={props.checkAnswer} inputNumber={props.inputNumber} actualAnswer={props.actualAnswer} setCompareAnswer={props.setCompareAnswer} />
    </div>
  )
}