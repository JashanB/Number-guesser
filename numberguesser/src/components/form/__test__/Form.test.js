import React from "react";
import ReactDOM from 'react-dom';
import Form from './../Form';
import Button from './../Form';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

const setup = () => {
  const form = render(<Form inputNumber={5}></Form>);
  const input = form.getByTestId('number-input');
  return {
    input,
    ...form
  }
}
afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Form></Form>, div)
})

it ("renders button correctly", () => {
  const {getByTestId} = render(<Button></Button>);
  expect(getByTestId('check-button')).toHaveTextContent("Check!")
})

it ("renders form correctly with passed down input", () => {
  const form = render(<Form inputNumber={5}></Form>);
  const input = form.getByTestId('number-input');
  expect(input.value).toBe("5")
})

// it ("changes input value", () => {
//   const form = render(<Form inputNumber={5}></Form>);
//   const input = form.getByTestId('number-input');
//   fireEvent.change(input, {target: {value: "10"}})
//   expect(form).toHaveValue("10")
// })
it("matches snapshot", () => {
  const tree = renderer.create(<Form inputNumber={5}></Form>).toJSON();
  expect(tree).toMatchSnapshot();
})