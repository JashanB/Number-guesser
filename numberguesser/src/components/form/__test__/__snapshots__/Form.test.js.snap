// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="input-your-guess"
>
  <form
    autoComplete="off"
    data-testid="input-form"
    onSubmit={[Function]}
  >
    <input
      className="guess-number"
      data-testid="number-input"
      onChange={[Function]}
      placeholder="Guess a number!"
      type="number"
      value={5}
    />
  </form>
  <button
    className="check-button"
    data-testid="check-button"
    onClick={[Function]}
  >
    Check!
  </button>
</div>
`;
